src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ExpressionHelper.cs

```diff
@@ -21,11 +21,26 @@ public static string GetExpressionText(string expression)
 
         public static string GetExpressionText(LambdaExpression expression)
         {
+            return GetExpressionText(expression, null);
+        }
+
+        public static string GetExpressionText(LambdaExpression expression, ExpressionTextCache expressionTextCache)
+        {
             if (expression == null)
             {
                 throw new ArgumentNullException(nameof(expression));
             }
 
+            string expressionText;
+            if (expressionTextCache != null &&
+                expressionTextCache.Entries.TryGetValue(expression, out expressionText))
+            {
+                return expressionText;
+            }
+
+            expressionText = string.Empty;
```

Comment:
Too far from where it's used. Do this in an `else` block for the `nameParts.Count > 0` condition.